cmd/dfget/app/root.go

```diff
@@ -0,0 +1,217 @@
+package app
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+
+	cfg "github.com/dragonflyoss/Dragonfly/dfget/config"
+	"github.com/dragonflyoss/Dragonfly/dfget/core"
+	"github.com/dragonflyoss/Dragonfly/dfget/errors"
+	"github.com/dragonflyoss/Dragonfly/dfget/util"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var (
+	localLimit string
+	totalLimit string
+	filter     string
+)
+
+var rootCmd = &cobra.Command{
+	Use:   "dfget",
+	Short: "The dfget is the client of Dragonfly.",
+	Long:  "The dfget is the client of Dragonfly, a non-interactive P2P downloader.",
+	Run: func(cmd *cobra.Command, args []string) {
+		checkParameters()
+		initLog()
+		cfg.AssertContext(cfg.Ctx)
+		cfg.Ctx.ClientLogger.Infof("cmd params:%q", args)
+		initProperties()
+		cfg.Ctx.ClientLogger.Infof("context:%s", cfg.Ctx)
+		err := core.Start(cfg.Ctx)
+		if err != nil {
+			util.Printer.Println(resultMsg(cfg.Ctx, time.Now(), err))
+			os.Exit(err.Code)
+		}
+		util.Printer.Println(resultMsg(cfg.Ctx, time.Now(), err))
+		os.Exit(0)
+	},
+}
+
+func init() {
+	initFlags()
+}
+
+func checkParameters() {
+	fmt.Println(os.Args)
+	if len(os.Args) < 3 {
```

Comment:
Just out of curiosity, why change `2` to `3`?

Reply:
When I type the commands like this `dfget -u`, the `len(os.Args)` is 2. That is illegal input. For dfget, We must supply a valid URL.

Reply:
However, there are also parameters of bool.

Reply:
Please give me an example which can prove `len(os.Args) < 2` is right

Reply:
I see. I will modify it.